### PR TITLE
Add test case of unjoining not ready cluster

### LINKED ISSUE #1055: [E2E]: Add test case of unjoining not ready cluster
**What would you like to be added**:
To surport unjoining the not ready member cluster, #1053 introduced a feature that abort deleting workload when the member cluster is unready. But the test case of E2E still missing.

**Why is this needed**:

### PR DESCRIPTION
**Pull request by jameszhangyukun**
**What type of PR is this?**

/kind feature
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #1055 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: test/e2e/namespace_test.go [hunk 1]
```diff
@@ -139,4 +139,107 @@ var _ = ginkgo.Describe("[namespace auto-provision] namespace auto-provision tes
...
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+				cluster.Status.Conditions = []metav1.Condition{
+					{
+						Status: "False",
+					},
+				}
+				_, err = karmadaClient.ClusterV1alpha1().Clusters().Update(context.TODO(), cluster, metav1.UpdateOptions{})
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			})
+		})
```
Comment (XiShanYongYe-Chang) [inline test/e2e/namespace_test.go:193]: Only update the conditions maybe can not cover the case, we can call `disableCluster` to disable cluster:

https://github.com/karmada-io/karmada/blob/697e120922db5b67212308ba9a3a39810a896096/test/e2e/failover_test.go#L142-L143

File: test/e2e/namespace_test.go [hunk 2]
```diff
@@ -139,4 +139,107 @@ var _ = ginkgo.Describe("[namespace auto-provision] namespace auto-provision tes
...
+				}
+				err := karmadactl.RunJoin(os.Stdout, karmadaConfig, opts)
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			})
+		})
+		ginkgo.BeforeEach(func() {
+			ginkgo.By(fmt.Sprintf("Creating Deployment workload in cluster: %s", clusterName), func() {
+
+				deployment := helper.NewDeployment(testNamespace, deploymentName)
+				framework.CreateDeployment(kubeClient, deployment)
```
Comment (XiShanYongYe-Chang) [inline test/e2e/namespace_test.go:177]: We also need to create a PropagationPolicy to propagate Deployment, for example:
https://github.com/karmada-io/karmada/blob/697e120922db5b67212308ba9a3a39810a896096/test/e2e/resource_test.go#L40-L41

**Review by XiShanYongYe-Chang — COMMENTED** [review]
This case tests the removal of a cluster, we can move this case to a new file.

We can test the code of a module locally. For example, we test the case which name contains `BasicPropagation`:

```
ginkgo -v -race -failFast -focus="BasicPropagation" ./test/e2e/
```

File: test/e2e/unjoin_test.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	"k8s.io/client-go/tools/clientcmd"
+
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/karmadactl"
+	"github.com/karmada-io/karmada/pkg/karmadactl/options"
+	"github.com/karmada-io/karmada/test/e2e/framework"
+	"github.com/karmada-io/karmada/test/helper"
+)
+
+var _ = ginkgo.Describe("[namespace auto-provision] namespace auto-provision testing", func() {
```
Comment (XiShanYongYe-Chang) [inline test/e2e/unjoin_test.go:21]: This describe text needs to be updated.

File: test/e2e/unjoin_test.go [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+		deployment := helper.NewDeployment(deploymentNamespace, deploymentName)
+		policy := helper.NewPropagationPolicy(policyNamespace, policyName, []policyv1alpha1.ResourceSelector{
+			{
+				APIVersion: deployment.APIVersion,
+				Kind:       deployment.Kind,
+				Name:       deployment.Name,
+			},
+		}, policyv1alpha1.Placement{
+			ClusterAffinity: &policyv1alpha1.ClusterAffinity{
+				ClusterNames: framework.ClusterNames(),
```
Comment (XiShanYongYe-Chang) [inline test/e2e/unjoin_test.go:43]: Here can directly assign `clusterName`:
```go
ClusterNames: []string{clusterName},
```

File: test/e2e/unjoin_test.go [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+					ClusterContext:    clusterContext,
+					ClusterKubeConfig: kubeConfigPath,
+				}
+				err := karmadactl.RunJoin(os.Stdout, karmadaConfig, opts)
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			})
+		})
+		ginkgo.BeforeEach(func() {
+			ginkgo.By(fmt.Sprintf("Creating Deployment workload in cluster: %s", clusterName), func() {
+
```
Comment (XiShanYongYe-Chang) [inline test/e2e/unjoin_test.go:72]: Delete the blank lines. Same as below.

File: test/e2e/unjoin_test.go [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+				Wait:              options.DefaultKarmadactlCommandDuration,
+			}
+			err := karmadactl.RunUnjoin(os.Stdout, karmadaConfig, opts)
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			err = deleteCluster(clusterName, kubeConfigPath)
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+			deployment, err := kubeClient.AppsV1().Deployments(testNamespace).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			gomega.Expect(deployment).ShouldNot(gomega.BeNil())
```
Comment (XiShanYongYe-Chang) [inline test/e2e/unjoin_test.go:105]: It's just a question. I don't understand why we need to query Deployment here.

Comment (jameszhangyukun) [reply test/e2e/unjoin_test.go:105]: I want to make sure whether the workload has not been deleted When unjoining not ready cluster

Comment (XiShanYongYe-Chang) [reply test/e2e/unjoin_test.go:99]: ok, I get it.

**Comment by XiShanYongYe-Chang** [comment]
@jameszhangyukun thanks :)

I think this pr will work after #1053 has been merged.

**Comment by RainbowMango** [comment]
OK. I‘ll re-trigger it after #1053

**Comment by XiShanYongYe-Chang** [comment]
hi, @jameszhangyukun #1053 has been merged, maybe you need to rebase master branch.

**Comment by RainbowMango** [comment]
Yeah, re-trigger does not help.

**Comment by jameszhangyukun** [comment]
Sorry for all, I  make a mistake operation deleted the old branch, so re-trigger does not help. I will reopen a new pr

**Pull request closed without merging**